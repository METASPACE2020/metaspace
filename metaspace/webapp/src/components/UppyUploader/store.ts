// based on https://uppy.io/docs/stores/#Implementing-Stores

import { ref } from 'vue'
import { State } from '@uppy/core'

type Listener = (prevState: State, nextState: State, patch: object) => void

export default () => {
  const state = ref<State | any>({})
  const listeners = new Set<Listener>()

  return {
    getState: () => state.value,
    setState: (patch: object) => {
      const prevState = state.value
      const nextState = { ...prevState, ...patch }

      state.value = nextState

      listeners.forEach((listener) => {
        listener(prevState, nextState, patch)
      })
    },
    subscribe: (listener: Listener) => {
      listeners.add(listener)
      return () => listeners.delete(listener)
    },
  }
}
