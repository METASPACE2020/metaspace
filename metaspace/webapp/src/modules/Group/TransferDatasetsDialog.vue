<template>
  <el-dialog
    class="dialog"
    :model-value="true"
    append-to-body
    :title="allDatasets.length > 0 ? 'Transfer datasets' : 'Join group'"
    @close="handleClose"
  >
    <div v-loading="loading">
      <div v-if="allDatasets.length > 0">
        <p style="margin-top: 0">
          You have previously uploaded one or more datasets without a group. Do you want to transfer any of these
          datasets to {{ groupName }}?
        </p>
        <dataset-checkbox-list v-model="selectedDatasets" :datasets="allDatasets" :init-select-all="true" />
        <p v-if="!isInvited">An email will be sent to the group's principal investigator to confirm your access.</p>
        <div class="button-bar">
          <el-button :disabled="isSubmitting" @click="handleClose"> Cancel </el-button>
          <el-button type="primary" :loading="isSubmitting" @click="handleAccept">
            {{ acceptText }}
          </el-button>
        </div>
      </div>
      <div v-else>
        <p v-if="!isInvited" style="margin-top: 0">
          An email will be sent to the group's principal investigator to confirm your access.
        </p>
        <p v-else style="margin-top: 0">Are you sure you want to join {{ groupName }}?</p>
        <div class="button-bar">
          <el-button :disabled="isSubmitting" size="small" @click="handleClose"> Cancel </el-button>
          <el-button type="primary" :loading="isSubmitting" size="small" @click="handleAccept">
            {{ acceptText }}
          </el-button>
        </div>
      </div>
    </div>
  </el-dialog>
</template>

<script lang="ts">
import { defineComponent, ref, computed } from 'vue'
import { useQuery } from '@vue/apollo-composable'
import { DatasetListItem, datasetListItemsQuery } from '../../api/dataset'
import DatasetCheckboxList from '../../components/DatasetCheckboxList.vue'
import { currentUserIdQuery } from '../../api/user'
import { ElDialog, ElButton, ElLoading } from '../../lib/element-plus'

export default defineComponent({
  components: {
    DatasetCheckboxList,
    ElDialog,
    ElButton,
  },
  directives: {
    loading: ElLoading.directive,
  },
  props: {
    groupName: {
      type: String,
      required: true,
    },
    isInvited: {
      type: Boolean,
      required: true,
    },
  },
  setup(props, { emit }) {
    const isSubmitting = ref(false)
    const selectedDatasets = ref({})

    const { result: currentUserResult, loading: currenUserLoading } = useQuery(currentUserIdQuery, null, {
      fetchPolicy: 'cache-first',
    })
    const currentUser = computed(() => currentUserResult.value?.currentUser)
    const { result: allDatasetsResult, loading: datasetsLoading } = useQuery(
      datasetListItemsQuery,
      () => ({
        dFilter: {
          submitter: currentUser.value ? currentUser.value.id : null,
          hasGroup: false,
        },
      }),
      {
        enabled: computed(() => !!currentUser.value && !!currentUser.value.id),
      }
    )
    const allDatasets = computed(() => (allDatasetsResult.value?.allDatasets as DatasetListItem[]) || [])
    const loading = computed(() => currenUserLoading.value || datasetsLoading.value)
    const dialogTitle = computed(() => (allDatasets.value.length > 0 ? 'Transfer datasets' : 'Join group'))
    const acceptText = computed(() => {
      const action = props.isInvited ? 'Join group' : 'Request access'
      const numSelected = Object.values(selectedDatasets.value).filter((selected) => selected).length
      return numSelected === 0
        ? action
        : `${action} and transfer ${numSelected} ${numSelected === 1 ? 'dataset' : 'datasets'}`
    })

    const handleAccept = () => {
      const selectedDatasetIds = Object.keys(selectedDatasets.value).filter((key) => selectedDatasets.value[key])
      emit('accept', selectedDatasetIds)
      isSubmitting.value = true
    }

    const handleClose = () => {
      if (!isSubmitting.value) {
        emit('close')
      }
    }

    return {
      loading,
      isSubmitting,
      selectedDatasets,
      currentUser,
      allDatasets,
      dialogTitle,
      acceptText,
      handleAccept,
      handleClose,
    }
  },
})
</script>
<style scoped lang="scss">
.dialog ::v-deep(.el-dialog) {
  @apply max-w-lg;
}
.button-bar {
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-end;
  margin-top: 20px;
}
</style>
