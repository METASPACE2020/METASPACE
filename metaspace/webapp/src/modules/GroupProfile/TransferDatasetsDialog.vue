<template>
    <el-dialog class="dialog"
               visible
               :title="allDatasets.length > 0 ? 'Transfer datasets' : 'Join group'"
               @close="handleClose">
      <div v-loading="loading">
        <div v-if="allDatasets.length > 0">
          <p style="margin-top: 0">
            You have previously uploaded one or more datasets without a group. Do you want to transfer any of these datasets to {{groupName}}?
          </p>
          <div class="dataset-list">
            <div v-for="dataset in allDatasets">
              <el-checkbox v-model="selectedDatasets[dataset.id]">
                {{dataset.name}}
                <span class="reset-color">(Submitted {{ formatDate(dataset.uploadDT) }})</span>
              </el-checkbox>
            </div>
          </div>
          <div class="select-buttons">
            <a href="#" @click.prevent="handleSelectNone">Select none</a>
            <span> | </span>
            <a href="#" @click.prevent="handleSelectAll">Select all</a>
          </div>
          <p v-if="!isInvited">
            An email will be sent to the group's principal investigator to confirm your access.
          </p>
          <div class="button-bar">
            <el-button :disabled="isSubmitting" @click="handleClose">Cancel</el-button>
            <el-button type="primary" :loading="isSubmitting" @click="handleAccept">{{acceptText}}</el-button>
          </div>
        </div>
        <div v-else>
          <p v-if="!isInvited" style="margin-top: 0">
            An email will be sent to the group's principal investigator to confirm your access.
          </p>
          <p v-else style="margin-top: 0">
            Are you sure you want to join {{groupName}}?
          </p>
          <div class="button-bar">
            <el-button :disabled="isSubmitting" @click="handleClose">Cancel</el-button>
            <el-button type="primary" :loading="isSubmitting" @click="handleAccept">{{acceptText}}</el-button>
          </div>
        </div>
      </div>
    </el-dialog>
</template>
<script lang="ts">
  import Vue from 'vue';
  import { Component, Prop, Watch } from 'vue-property-decorator';
  import { DatasetListItem, datasetListItemsQuery } from '../../api/dataset';
  import { fromPairs } from 'lodash-es';
  import format from 'date-fns/format';

  @Component({
    apollo: {
      allDatasets: {
        query: datasetListItemsQuery,
        loadingKey: 'loading',
        variables(this: TransferDatasetsDialog) {
          return {
            dFilter: {
              submitter: this.currentUserId,
              hasGroup: false,
            }
          };
        }
      }
    }
  })
  export default class TransferDatasetsDialog extends Vue {
    @Prop({ required: true })
    currentUserId!: string; //TODO: It's weird to have this as a prop. It would be better to grab it from Vuex.
    @Prop({ required: true })
    groupName!: string;
    @Prop({ required: true })
    isInvited!: boolean; // True to show "Accept/reject invitation", false to show "Request access/cancel"

    loading: number = 0;
    allDatasets: DatasetListItem[] = [];
    selectedDatasets: Record<string, boolean> = {};
    isSubmitting: Boolean = false;

    get numSelected() {
      return Object.values(this.selectedDatasets).filter(selected => selected).length;
    }

    get acceptText() {
      const action = this.isInvited ? 'Join group' : 'Request access';
      return this.numSelected === 0
        ? action
        : `${action} and transfer ${this.numSelected} ${this.numSelected === 1 ? 'dataset' : 'datasets'}`
    }

    formatDate(date: string) {
      return `${format(date, 'YYYY-MM-DD')} at ${format(date, 'HH:mm')}`;
    }

    @Watch('allDatasets')
    populateSelectedDatasets() {
      //Rebuild `selectedDatasets` so that the keys are in sync with the ids from `datasets`
      this.selectedDatasets = fromPairs(this.allDatasets.map(({id}) => {
        return [id, id in this.selectedDatasets ? this.selectedDatasets[id] : true];
      }));
    }

    handleClose() {
      if (!this.isSubmitting) {
        this.$emit('close');
      }
    }

    handleSelectNone() {
      Object.keys(this.selectedDatasets).forEach(key => this.selectedDatasets[key] = false);
    }

    handleSelectAll() {
      Object.keys(this.selectedDatasets).forEach(key => this.selectedDatasets[key] = true);
    }

    handleAccept() {
      const selectedDatasetIds = Object.keys(this.selectedDatasets).filter(key => this.selectedDatasets[key]);
      this.$emit('accept', selectedDatasetIds);
      this.isSubmitting = true;
    }
  }
</script>
<style scoped lang="scss">
  @import "~element-ui/packages/theme-chalk/src/common/var";
  .dialog /deep/ .el-dialog {
    max-width: 600px;
  }
  .dataset-list {
    margin: 20px;
    max-height: 50vh;
    overflow: auto;
  }
  .select-buttons {
    text-align: right;
    margin: 20px;
  }
  .button-bar {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-end;
    margin-top: 20px;
  }
  .reset-color {
    color: $--color-text-regular;
  }

</style>
