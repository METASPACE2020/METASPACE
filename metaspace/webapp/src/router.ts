/* eslint-disable vue/max-len */
import Vue, { AsyncComponent } from 'vue'
import VueRouter, { RawLocation } from 'vue-router'
import AboutPage from './modules/App/AboutPage.vue'
import DatasetsPage from './modules/Datasets/DatasetsPage.vue'
import { DialogPage, ResetPasswordPage } from './modules/Account'
import { redirectAfterSignIn } from './modules/Account/signInReturnUrl'
import { updateDBParam } from './modules/Filters/url'
import NotFoundPage from './modules/App/NotFoundPage.vue'

Vue.use(VueRouter)

const asyncPagesFreelyTyped = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ './modules/Annotations/AnnotationsPage.vue'),
  DatasetSummary: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetSummary" */ './modules/Datasets/summary/DatasetSummary.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ './modules/MetadataEditor/MetadataEditPage.vue'),
  ImageAlignmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "ImageAlignmentPage" */ './modules/ImageAlignment/ImageAlignmentPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ './modules/MetadataEditor/UploadPage.vue'),
  DesignPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DesignPage" */ './design/DesignPage.vue'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/HelpPage.vue'),
  EditUserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/UserProfile/EditUserPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/GroupProfile/CreateGroupPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Project/ProjectsListPage.vue'),
  SystemHealthPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/SystemHealthPage.vue'),
  GroupsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/GroupsListPage.vue'),
  PrivacyPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PrivacyPage.vue'),
  TermsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/TermsPage.vue'),
  PublicationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PublicationsPage.vue'),

  // These pages use sanitizeHtml, which is big
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/GroupProfile/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Project/ViewProjectPage.vue'),
}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, AsyncComponent>

const convertLegacyUrls = () => {
  const { pathname, hash, search } = window.location
  if (pathname === '/' && hash && hash.startsWith('#/')) {
    history.replaceState(undefined, undefined as any, hash.slice(1))
  }
  if (pathname === '/annotations') {
    const updatedQueryString = updateDBParam(search.slice(1))
    if (updatedQueryString !== null) {
      history.replaceState(undefined, undefined as any, `${pathname}?${updatedQueryString}`)
    }
  }
}
convertLegacyUrls()

const router = new VueRouter({
  mode: 'history',
  routes: [
    { path: '/(about)?', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
    { path: '/annotations', component: asyncPages.AnnotationsPage },
    {
      path: '/datasets',
      component: DatasetsPage,
      children: [
        { path: '', component: asyncPages.DatasetTable },
        { path: 'summary', component: asyncPages.DatasetSummary },
      ],
    },
    { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
    { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.ImageAlignmentPage },
    { path: '/upload', component: asyncPages.UploadPage },
    { path: '/help', component: asyncPages.HelpPage, meta: { footer: true } },
    { path: '/user/me', component: asyncPages.EditUserPage },

    { path: '/admin/health', component: asyncPages.SystemHealthPage },
    { path: '/admin/groups', component: asyncPages.GroupsListPage },

    { path: '/account/sign-in', component: DialogPage, props: { dialog: 'signIn' } },
    { path: '/account/sign-in-success', redirect: redirectAfterSignIn },
    { path: '/account/create-account', component: DialogPage, props: { dialog: 'createAccount' } },
    { path: '/account/forgot-password', component: DialogPage, props: { dialog: 'forgotPassword' } },
    { path: '/account/reset-password', component: ResetPasswordPage },

    { path: '/group/create', component: asyncPages.CreateGroupPage },
    { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },

    { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
    { // Legacy URL sent in "request access" emails up until Feb 2019
      path: '/project/:projectIdOrSlug/manage',
      redirect: { path: '/project/:projectIdOrSlug', query: { tab: 'members' } } as RawLocation,
    },
    { path: '/projects', component: asyncPages.ProjectsListPage },

    { path: '/terms', component: asyncPages.TermsPage, meta: { footer: true } },
    { path: '/privacy', component: asyncPages.PrivacyPage, meta: { footer: true } },
    { path: '/publications', component: asyncPages.PublicationsPage, meta: { footer: true } },

    { path: '/design', component: asyncPages.DesignPage, meta: { footer: true, flex: true } },

    { path: '*', component: NotFoundPage, meta: { footer: true, flex: true } },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (to.hash) {
      return { selector: to.hash, offset: { x: 0, y: 64 } } // offset header height
    }
    if (savedPosition) {
      return savedPosition
    }
    if (to.path !== from.path) {
      return { x: 0, y: 0 }
    }
  },
})

const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined, true)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
