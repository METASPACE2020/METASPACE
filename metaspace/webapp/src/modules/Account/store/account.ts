import { Module } from 'vuex';
import { DialogType } from '../dialogs';
import router from '../../../router';

export interface AccountState {
  dialog: DialogType | null;
  showDialogAsPage: boolean;
}

const account = {
  namespaced: true,
  state: {
    dialog: null,
    showDialogAsPage: false
  },
  mutations: {
    showDialog(state: AccountState, dialog: DialogType) {
      state.dialog = dialog;
      state.showDialogAsPage = false;
    },
    showDialogAsPage(state: AccountState, dialog: DialogType) {
      state.dialog = dialog;
      state.showDialogAsPage = true;
    },
    hideDialog(state: AccountState, dialog: DialogType) {
      if (state.dialog === dialog) {
        state.dialog = null;
        if (state.showDialogAsPage) {
          router.push('/');
        }
      }
    }
  }
};

export default account as Module<AccountState, any>;
