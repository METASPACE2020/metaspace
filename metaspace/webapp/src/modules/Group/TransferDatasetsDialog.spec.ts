import { flushPromises, mount } from '@vue/test-utils'
import { ElButton, ElCheckbox } from '../../lib/element-plus'
import { nextTick, ref } from 'vue'
import TransferDatasetsDialog from './TransferDatasetsDialog.vue'
import router from '../../router'
import { initMockGraphqlClient } from '../../tests/utils/mockGraphqlClient'
import { DefaultApolloClient, useQuery } from '@vue/apollo-composable'
import { vi } from 'vitest'
import store from '../../store'

let graphqlMocks

const mockDatasets = [
  { id: '2018-06-28_13h23m10s', name: 'Untreated_3_434', uploadDT: '2018-06-28T13:23:10.837000' },
  { id: '2018-06-28_13h21m36s', name: 'Dataset 2', uploadDT: '2018-06-28T13:21:36.973867' },
  { id: '2018-06-28_12h03m36s', name: 'Dataset 3', uploadDT: '2018-06-28T12:03:36.409743' },
  { id: '2018-06-28_13h20m44s', name: 'Dataset 4', uploadDT: '2018-06-28T13:20:44.736472' },
]
const mockProps = {
  currentUserId: 'current user id',
  groupName: 'Group Name',
  isInvited: true,
}
const stubs = {
  DatasetItem: {
    template: '<div class="mock-ds-item"><slot></slot></div>',
    props: ['dataset'],
  },
  ElTable: {
    template: '<div class="mock-el-table"><slot></slot></div>',
    props: ['data', 'columns', 'tableId'], // include other props as necessary
  },
  ElTableColumn: {
    template: '<div class="mock-el-table-column"><slot></slot></div>',
    props: ['data', 'columns', 'tableId'], // include other props as necessary
  },
}

vi.mock('@vue/apollo-composable', () => ({
  useQuery: vi.fn(),
  DefaultApolloClient: vi.fn(),
}))
const mockGraphql = async (params) => {
  graphqlMocks = await initMockGraphqlClient({
    Query: () => params,
  })
  ;(useQuery as any).mockReturnValue({
    result: ref(Object.keys(params).reduce((acc, key) => ({ ...acc, [key]: params[key]() }), {})),
    loading: ref(false),
    onResult: vi.fn(),
  })
}
describe('TransferDatasetsDialog', () => {
  ;[false, true].forEach((hasDatasets) => {
    ;[false, true].forEach((isInvited) => {
      it(`should match snapshot (${hasDatasets ? 'datasets to import' : 'no datasets'}, ${
        isInvited ? 'invited' : 'requesting access'
      })`, async () => {
        if (hasDatasets) {
          await mockGraphql({ allDatasets: () => mockDatasets })
        } else {
          await mockGraphql({ allDatasets: () => [] })
        }
        const props = { ...mockProps, isInvited }
        const wrapper = mount(TransferDatasetsDialog, {
          global: {
            plugins: [store, router],
            stubs,
            provide: {
              [DefaultApolloClient]: graphqlMocks,
            },
          },
          props,
        })

        await flushPromises()
        await nextTick()

        expect(wrapper.html()).toMatchSnapshot()
      })
    })
  })

  it('should call back on success when some datasets are selected', async () => {
    await mockGraphql({ allDatasets: () => mockDatasets })
    const wrapper = mount(TransferDatasetsDialog, {
      global: {
        plugins: [store, router],
        stubs,
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
      props: mockProps,
    })
    await flushPromises()
    await nextTick()

    const checkbox = wrapper.findComponent(ElCheckbox)
    await checkbox.trigger('click')
    await nextTick()

    const primaryButton = wrapper
      .findAllComponents(ElButton)
      .filter((b) => b.props().type === 'primary')
      .at(0)
    await primaryButton.trigger('click')
    await nextTick()

    expect(wrapper.emitted('accept')).toEqual([[mockDatasets.slice(1).map((ds) => ds.id)]])
  })
})
