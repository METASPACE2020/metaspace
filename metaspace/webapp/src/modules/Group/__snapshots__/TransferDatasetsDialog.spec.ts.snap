// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TransferDatasetsDialog > should match snapshot (datasets to import, invited) 1`] = `
"<div data-v-28cb04fb=\\"\\" class=\\"mock-el-dialog dialog\\" model-value=\\"true\\" append-to-body=\\"\\" title=\\"Transfer datasets\\" currentuserid=\\"current user id\\">
  <div data-v-28cb04fb=\\"\\">
    <div data-v-28cb04fb=\\"\\">
      <p data-v-28cb04fb=\\"\\" style=\\"margin-top: 0px;\\"> You have previously uploaded one or more datasets without a group. Do you want to transfer any of these datasets to Group Name? </p>
      <div data-v-c3668e62=\\"\\" data-v-28cb04fb=\\"\\" modelvalue=\\"[object Object]\\">
        <div data-v-c3668e62=\\"\\" class=\\"select-buttons\\"><a data-v-c3668e62=\\"\\" href=\\"#\\">Select none</a><span data-v-c3668e62=\\"\\"> | </span><a data-v-c3668e62=\\"\\" href=\\"#\\">Select all</a></div>
        <div data-v-c3668e62=\\"\\" class=\\"dataset-checkbox-list leading-6\\">
          <div data-v-c3668e62=\\"\\"><label data-v-c3668e62=\\"\\" class=\\"el-checkbox is-checked flex h-6 items-center\\"><span class=\\"el-checkbox__input is-checked\\"><input class=\\"el-checkbox__original\\" type=\\"checkbox\\"><span class=\\"el-checkbox__inner\\"></span></span><span class=\\"el-checkbox__label\\"><span data-v-c3668e62=\\"\\" class=\\"truncate\\" title=\\"Untreated_3_434\\">Untreated_3_434</span><span data-v-c3668e62=\\"\\" class=\\"text-gray-700 text-xs tracking-wide pl-1\\"><span data-v-c3668e62=\\"\\" class=\\"sm-elapsed-time\\" title=\\"2018/06/28, 13:23\\">2 years ago</span></span>
              <!--v-if--></span>
            </label></div>
          <div data-v-c3668e62=\\"\\"><label data-v-c3668e62=\\"\\" class=\\"el-checkbox is-checked flex h-6 items-center\\"><span class=\\"el-checkbox__input is-checked\\"><input class=\\"el-checkbox__original\\" type=\\"checkbox\\"><span class=\\"el-checkbox__inner\\"></span></span><span class=\\"el-checkbox__label\\"><span data-v-c3668e62=\\"\\" class=\\"truncate\\" title=\\"Dataset 2\\">Dataset 2</span><span data-v-c3668e62=\\"\\" class=\\"text-gray-700 text-xs tracking-wide pl-1\\"><span data-v-c3668e62=\\"\\" class=\\"sm-elapsed-time\\" title=\\"2018/06/28, 13:21\\">2 years ago</span></span>
              <!--v-if--></span>
            </label></div>
          <div data-v-c3668e62=\\"\\"><label data-v-c3668e62=\\"\\" class=\\"el-checkbox is-checked flex h-6 items-center\\"><span class=\\"el-checkbox__input is-checked\\"><input class=\\"el-checkbox__original\\" type=\\"checkbox\\"><span class=\\"el-checkbox__inner\\"></span></span><span class=\\"el-checkbox__label\\"><span data-v-c3668e62=\\"\\" class=\\"truncate\\" title=\\"Dataset 3\\">Dataset 3</span><span data-v-c3668e62=\\"\\" class=\\"text-gray-700 text-xs tracking-wide pl-1\\"><span data-v-c3668e62=\\"\\" class=\\"sm-elapsed-time\\" title=\\"2018/06/28, 12:03\\">2 years ago</span></span>
              <!--v-if--></span>
            </label></div>
          <div data-v-c3668e62=\\"\\"><label data-v-c3668e62=\\"\\" class=\\"el-checkbox is-checked flex h-6 items-center\\"><span class=\\"el-checkbox__input is-checked\\"><input class=\\"el-checkbox__original\\" type=\\"checkbox\\"><span class=\\"el-checkbox__inner\\"></span></span><span class=\\"el-checkbox__label\\"><span data-v-c3668e62=\\"\\" class=\\"truncate\\" title=\\"Dataset 4\\">Dataset 4</span><span data-v-c3668e62=\\"\\" class=\\"text-gray-700 text-xs tracking-wide pl-1\\"><span data-v-c3668e62=\\"\\" class=\\"sm-elapsed-time\\" title=\\"2018/06/28, 13:20\\">2 years ago</span></span>
              <!--v-if--></span>
            </label></div>
        </div>
      </div>
      <!--v-if-->
      <div data-v-28cb04fb=\\"\\" class=\\"button-bar\\"><button data-v-28cb04fb=\\"\\" ariadisabled=\\"false\\" type=\\"button\\" class=\\"el-button\\">
          <!--v-if--><span class=\\"\\"> Cancel </span>
        </button><button data-v-28cb04fb=\\"\\" ariadisabled=\\"false\\" type=\\"button\\" class=\\"el-button el-button--primary\\">
          <!--v-if--><span class=\\"\\">Join group</span>
        </button></div>
    </div>
  </div>
</div>"
`;

exports[`TransferDatasetsDialog > should match snapshot (datasets to import, requesting access) 1`] = `
"<div data-v-28cb04fb=\\"\\" class=\\"mock-el-dialog dialog\\" model-value=\\"true\\" append-to-body=\\"\\" title=\\"Transfer datasets\\" currentuserid=\\"current user id\\">
  <div data-v-28cb04fb=\\"\\">
    <div data-v-28cb04fb=\\"\\">
      <p data-v-28cb04fb=\\"\\" style=\\"margin-top: 0px;\\"> You have previously uploaded one or more datasets without a group. Do you want to transfer any of these datasets to Group Name? </p>
      <div data-v-c3668e62=\\"\\" data-v-28cb04fb=\\"\\" modelvalue=\\"[object Object]\\">
        <div data-v-c3668e62=\\"\\" class=\\"select-buttons\\"><a data-v-c3668e62=\\"\\" href=\\"#\\">Select none</a><span data-v-c3668e62=\\"\\"> | </span><a data-v-c3668e62=\\"\\" href=\\"#\\">Select all</a></div>
        <div data-v-c3668e62=\\"\\" class=\\"dataset-checkbox-list leading-6\\">
          <div data-v-c3668e62=\\"\\"><label data-v-c3668e62=\\"\\" class=\\"el-checkbox is-checked flex h-6 items-center\\"><span class=\\"el-checkbox__input is-checked\\"><input class=\\"el-checkbox__original\\" type=\\"checkbox\\"><span class=\\"el-checkbox__inner\\"></span></span><span class=\\"el-checkbox__label\\"><span data-v-c3668e62=\\"\\" class=\\"truncate\\" title=\\"Untreated_3_434\\">Untreated_3_434</span><span data-v-c3668e62=\\"\\" class=\\"text-gray-700 text-xs tracking-wide pl-1\\"><span data-v-c3668e62=\\"\\" class=\\"sm-elapsed-time\\" title=\\"2018/06/28, 13:23\\">2 years ago</span></span>
              <!--v-if--></span>
            </label></div>
          <div data-v-c3668e62=\\"\\"><label data-v-c3668e62=\\"\\" class=\\"el-checkbox is-checked flex h-6 items-center\\"><span class=\\"el-checkbox__input is-checked\\"><input class=\\"el-checkbox__original\\" type=\\"checkbox\\"><span class=\\"el-checkbox__inner\\"></span></span><span class=\\"el-checkbox__label\\"><span data-v-c3668e62=\\"\\" class=\\"truncate\\" title=\\"Dataset 2\\">Dataset 2</span><span data-v-c3668e62=\\"\\" class=\\"text-gray-700 text-xs tracking-wide pl-1\\"><span data-v-c3668e62=\\"\\" class=\\"sm-elapsed-time\\" title=\\"2018/06/28, 13:21\\">2 years ago</span></span>
              <!--v-if--></span>
            </label></div>
          <div data-v-c3668e62=\\"\\"><label data-v-c3668e62=\\"\\" class=\\"el-checkbox is-checked flex h-6 items-center\\"><span class=\\"el-checkbox__input is-checked\\"><input class=\\"el-checkbox__original\\" type=\\"checkbox\\"><span class=\\"el-checkbox__inner\\"></span></span><span class=\\"el-checkbox__label\\"><span data-v-c3668e62=\\"\\" class=\\"truncate\\" title=\\"Dataset 3\\">Dataset 3</span><span data-v-c3668e62=\\"\\" class=\\"text-gray-700 text-xs tracking-wide pl-1\\"><span data-v-c3668e62=\\"\\" class=\\"sm-elapsed-time\\" title=\\"2018/06/28, 12:03\\">2 years ago</span></span>
              <!--v-if--></span>
            </label></div>
          <div data-v-c3668e62=\\"\\"><label data-v-c3668e62=\\"\\" class=\\"el-checkbox is-checked flex h-6 items-center\\"><span class=\\"el-checkbox__input is-checked\\"><input class=\\"el-checkbox__original\\" type=\\"checkbox\\"><span class=\\"el-checkbox__inner\\"></span></span><span class=\\"el-checkbox__label\\"><span data-v-c3668e62=\\"\\" class=\\"truncate\\" title=\\"Dataset 4\\">Dataset 4</span><span data-v-c3668e62=\\"\\" class=\\"text-gray-700 text-xs tracking-wide pl-1\\"><span data-v-c3668e62=\\"\\" class=\\"sm-elapsed-time\\" title=\\"2018/06/28, 13:20\\">2 years ago</span></span>
              <!--v-if--></span>
            </label></div>
        </div>
      </div>
      <p data-v-28cb04fb=\\"\\">An email will be sent to the group's principal investigator to confirm your access.</p>
      <div data-v-28cb04fb=\\"\\" class=\\"button-bar\\"><button data-v-28cb04fb=\\"\\" ariadisabled=\\"false\\" type=\\"button\\" class=\\"el-button\\">
          <!--v-if--><span class=\\"\\"> Cancel </span>
        </button><button data-v-28cb04fb=\\"\\" ariadisabled=\\"false\\" type=\\"button\\" class=\\"el-button el-button--primary\\">
          <!--v-if--><span class=\\"\\">Request access</span>
        </button></div>
    </div>
  </div>
</div>"
`;

exports[`TransferDatasetsDialog > should match snapshot (no datasets, invited) 1`] = `
"<div data-v-28cb04fb=\\"\\" class=\\"mock-el-dialog dialog\\" model-value=\\"true\\" append-to-body=\\"\\" title=\\"Join group\\" currentuserid=\\"current user id\\">
  <div data-v-28cb04fb=\\"\\">
    <div data-v-28cb04fb=\\"\\">
      <p data-v-28cb04fb=\\"\\" style=\\"margin-top: 0px;\\">Are you sure you want to join Group Name?</p>
      <div data-v-28cb04fb=\\"\\" class=\\"button-bar\\"><button data-v-28cb04fb=\\"\\" ariadisabled=\\"false\\" type=\\"button\\" class=\\"el-button el-button--small\\">
          <!--v-if--><span class=\\"\\"> Cancel </span>
        </button><button data-v-28cb04fb=\\"\\" ariadisabled=\\"false\\" type=\\"button\\" class=\\"el-button el-button--primary el-button--small\\">
          <!--v-if--><span class=\\"\\">Join group</span>
        </button></div>
    </div>
  </div>
</div>"
`;

exports[`TransferDatasetsDialog > should match snapshot (no datasets, requesting access) 1`] = `
"<div data-v-28cb04fb=\\"\\" class=\\"mock-el-dialog dialog\\" model-value=\\"true\\" append-to-body=\\"\\" title=\\"Join group\\" currentuserid=\\"current user id\\">
  <div data-v-28cb04fb=\\"\\">
    <div data-v-28cb04fb=\\"\\">
      <p data-v-28cb04fb=\\"\\" style=\\"margin-top: 0px;\\"> An email will be sent to the group's principal investigator to confirm your access. </p>
      <div data-v-28cb04fb=\\"\\" class=\\"button-bar\\"><button data-v-28cb04fb=\\"\\" ariadisabled=\\"false\\" type=\\"button\\" class=\\"el-button el-button--small\\">
          <!--v-if--><span class=\\"\\"> Cancel </span>
        </button><button data-v-28cb04fb=\\"\\" ariadisabled=\\"false\\" type=\\"button\\" class=\\"el-button el-button--primary el-button--small\\">
          <!--v-if--><span class=\\"\\">Request access</span>
        </button></div>
    </div>
  </div>
</div>"
`;
